src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationOnSameLineCheck.java

```diff
@@ -0,0 +1,126 @@
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * The check does verifying that annotations are located on the same line with their targets.
+ * @author zenigata
+ */
+public class AnnotationOnSameLineCheck extends AbstractCheck {
+
+    /** A key is pointing to the warning message text in "messages.properties" file. */
+    public static final String MSG_KEY_ANNOTATION_ON_SAME_LINE = "annotation.same.line";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.VARIABLE_DEF,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.VARIABLE_DEF,
+            TokenTypes.PARAMETER_DEF,
+            TokenTypes.ANNOTATION_DEF,
+            TokenTypes.TYPECAST,
+            TokenTypes.LITERAL_THROWS,
+            TokenTypes.IMPLEMENTS_CLAUSE,
+            TokenTypes.TYPE_ARGUMENT,
+            TokenTypes.LITERAL_NEW,
+            TokenTypes.DOT,
+            TokenTypes.ANNOTATION_FIELD_DEF,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtils.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        DetailAST nodeWithAnnotations = ast;
+        if (ast.getType() == TokenTypes.TYPECAST) {
+            nodeWithAnnotations = ast.findFirstToken(TokenTypes.TYPE);
+        }
+        DetailAST modifiersNode = nodeWithAnnotations.findFirstToken(TokenTypes.MODIFIERS);
+        if (modifiersNode == null) {
+            modifiersNode = nodeWithAnnotations.findFirstToken(TokenTypes.ANNOTATIONS);
+        }
+        if (modifiersNode != null) {
+            for (DetailAST annotationNode = modifiersNode.getFirstChild();
+                    annotationNode != null;
+                    annotationNode = annotationNode.getNextSibling()) {
+                if (annotationNode.getType() == TokenTypes.ANNOTATION
+                        && annotationNode.getLineNo() != getNextNode(annotationNode).getLineNo()) {
+                    log(annotationNode.getLineNo(), MSG_KEY_ANNOTATION_ON_SAME_LINE,
+                          getAnnotationName(annotationNode));
+                }
+            }
+        }
+    }
+
+    /**
+     * Finds next node of ast tree.
+     * @param node current node
+     * @return node that is next to given
+     */
+    private static DetailAST getNextNode(DetailAST node) {
+        final DetailAST nextNode;
```

Comment:
All your other forms of writing this was writing the assignment with the standard one, and than change it's assignment if it was null. Why did you change it here?
````
DetailAST nextNode = node.getNextSibling();
if (nextNode == null) {
    nextNode = node.getParent().getNextSibling();
}
````

Reply:
changed